### PR TITLE
Added new VERBOSE logging level and fixed enclave logging issues

### PR DESCRIPTION
**Pull request by soccerGB**
This PR involves the following changes:

1. Added a new OE_LOG_LEVEL_VERBOSE level to the existing SDK logging mechanism
2. Addressed enclave logging issue
3. Moved some detailed INFO message to VERBOSE
4. Enhanced logging for attestation code paths

### CODE REVIEW

**Comment by soccerGB** [comment]
bors try

File: include/openenclave/internal/raise.h [hunk 1]
```diff
@@ -67,15 +67,15 @@
...
-            OE_TRACE_ERROR("Failed::%s", oe_result_str(result)); \
-        }                                                        \
-        goto done;                                               \
+#define OE_RAISE(RESULT, ...)                             \
+    do                                                    \
+    {                                                     \
+        result = (RESULT);                                \
+        if (result != OE_OK)                              \
+        {                                                 \
+            OE_TRACE_ERROR(":%s", oe_result_str(result)); \
```
Comment (johnkord) [inline include/openenclave/internal/raise.h]: Did you mean to leave a colon in the string here?  ":%s" as opposed to "%s"?

File: common/safecrt.c [hunk 2]
```diff
@@ -66,7 +66,7 @@ oe_result_t oe_memset_s(void* dst, size_t dst_size, int value, size_t num_bytes)
     volatile unsigned char* p = dst;
 
     if (dst == NULL)
-        OE_RAISE(OE_INVALID_PARAMETER);
+        OE_RAISE_NO_TRACE(OE_INVALID_PARAMETER);
```
Comment (anakrish) [inline common/safecrt.c]: It seems strange that general purpose CRT routines would decide to do no-trace

File: enclave/key.c [hunk 3]
```diff
@@ -323,7 +323,7 @@ oe_result_t oe_public_key_free(oe_public_key_t* public_key, uint64_t magic)
     if (public_key)
     {
         if (!oe_public_key_is_valid(public_key, magic))
-            OE_RAISE(OE_INVALID_PARAMETER);
+            OE_RAISE_NO_TRACE(OE_INVALID_PARAMETER);
```
Comment (anakrish) [inline enclave/key.c]: Again the rationale is not clear here. Why no trace?

File: include/openenclave/internal/raise.h (hunk 1 above)
Comment (soccerGB) [reply include/openenclave/internal/raise.h]: Having ":" there was on purpose.

File: enclave/key.c (hunk 3 above)
Comment (soccerGB) [reply enclave/key.c]: I was trying to remove unnecessary error messages: in some error cases , when jumping to "done" label, we ended up spewing confusing error message for those clean up operations.

File: common/safecrt.c (hunk 2 above)
Comment (soccerGB) [reply common/safecrt.c]: Good comment!
I was trying to remove unnecessary error messages but this change might have over-done it.
I will put them back.

**Pull request merged**